addon/instance-initializers/prefetch.js

```diff
@@ -38,7 +38,13 @@ export function initialize(instance) {
       }
 
       // Run the prefetch hook if the route has one.
-      handlerInfo.handler._prefetched = handlerInfo.runSharedModelHook(transition, 'prefetch', [fullParams]);
+      if (!handlerInfo.handler && handlerInfo.handlerPromise) {
+        handlerInfo.handlerPromise.then((handler) => {
+          handler._prefetched = handlerInfo.runSharedModelHook(transition, 'prefetch', [fullParams]);
+        });
+      } else {
+        handlerInfo.handler._prefetched = handlerInfo.runSharedModelHook(transition, 'prefetch', [fullParams]);
```

Comment:
That is dependent on the version of Ember you're using. The `handlerPromise` wasn't around prior to 2.7 or something.